code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -51,10 +53,20 @@
 	return
 
 /datum/reagent/proc/on_mob_life(mob/living/carbon/M, alien)
+	purge(M)
```

Comment:
medicine and consumable reagents completely override the parent's `on_mob_life`.

Reply:
Most of them reference the root on_mob_life